tabpy/tabpy_tools/client.py

```diff
@@ -22,14 +22,13 @@ def _check_endpoint_type(name):
 
 def _check_hostname(name):
     _check_endpoint_type(name)
-    hostname_checker = compile(
-        r'^http(s)?://[a-zA-Z0-9-_\.]+(/)?(:[0-9]+)?(/)?$')
+    hostname_checker = compile(r'^http(s)?://[\w.-]+(/)?(:\d+)?(/)?$')
```

Comment:
Just out of curiosity is there a reason to use this regex instead of `urllib.parse`?

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

Reply:
That's a good idea. It could add more functionality if needed but it's probably just regex under the hood right?

Reply:
I don't know the details but can find here:

https://github.com/python/cpython/blob/f501db2b93a9d3d840b6fb38d6bdda8bcc400d4a/Lib/urllib/parse.py#L366

unless needed to roll separately here I just think a better idea to use built-ins. One gap in this implementation is for ipv6 addresses, but ultimately up to the Tableau team

Reply:
I think we should prefer a dedicated URL parser/validator even if it is implemented with exactly the same regex.